components/brave_user_agent/browser/brave_user_agent_service.cc

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_user_agent/browser/brave_user_agent_service.h"
+
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/contains.h"
+#include "base/files/file_path.h"
+#include "base/strings/string_split.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/brave_user_agent/browser/brave_user_agent_component_installer.h"
+
+constexpr char kBraveUserAgentExceptionsFile[] = "brave-checks.txt";
+
+namespace brave_user_agent {
+
+BraveUserAgentService::BraveUserAgentService(
+    component_updater::ComponentUpdateService* cus)
+    : component_update_service_(cus) {
+  // Can be nullptr in unit tests
+  if (cus) {
+    RegisterBraveUserAgentComponent(
```

Comment:
This should go in `RegisterComponentsForUpdate`. The canonical pattern for component installers is to have a singleton that is accessed from the component installer to load the data. See `PsstRuleRegistry` for an example

Reply:
Updated to use singleton & register in `RegisterComponentsForUpdate` in https://github.com/brave/brave-core/pull/27620/commits/93adc7db80f45984cfeae079f8cbefec71c6140f